Documentation/ApiOverview/Ajax/Backend/Index.rst

```diff
@@ -0,0 +1,161 @@
+
+
+Create a Controller
+===================
+
+By convention, a controller is placed within the extension's :file:`Controller` directory, optionally in a subdirectory.
+To have such controller, create a new :php:`ExampleController` in :file:`Classes/Controller/ExampleController.php`
+inside your extension.
+
+The controller doesn't need that much logic right now. We'll create a method called :php:`doSomethingAction()` which
+will be our AJAX endpoint.
+
+.. code-block:: php
+
+   <?php
+   declare(strict_types = 1);
+
+   namespace Vendor\MyExtension\Controller;
+
+   use Psr\Http\Message\ServerRequestInterface;
+   use TYPO3\CMS\Core\Http\Response;
+
+   class ExampleController
+   {
+       public function doSomethingAction(ServerRequestInterface $request): Response
+       {
+       }
+   }
+
+
+In its current state, the method doesn't do anything yet. We can add a very generic handling that exponentiates an
+incoming number by 2. The incoming value will be passed as a query string argument named `input`.
+
+.. code-block:: php
+
+   public function doSomethingAction(ServerRequestInterface $request): Response
+   {
+       $input = $request->getQueryParams()['input'] ?? null;
+       if ($input === null) {
+           throw new \InvalidArgumentException('Please provide a number', 1580585107);
+       }
+
+       $result = $input ** 2;
+   }
+
+
+.. note::
+   This is a really simple example. Something like this should not be used in production, as such feature is available
+   in JavaScript as well.
+
+
+We have computed our result by using the `exponentiation operator`_, but we don't do anything with it yet. It's time to
+build a proper response. A response implements the :php:`Psr\Http\Message\ResponseInterface` and its constructor
+accepts the following arguments:
+
+.. rst-class:: dl-parameters
+
+$body
+   :sep:`|` :aspect:`Condition:` required
+   :sep:`|` :aspect:`Type:` string
+   :sep:`|`
+
+   The content of the response.
+
+$statusCode
+   :sep:`|` :aspect:`Condition:` optional
+   :sep:`|` :aspect:`Type:` int
+   :sep:`|` :aspect:`Default:` 200
+   :sep:`|`
+
+   The HTTP status code of the response. The default of `200` means `OK`.
+
+$headers
+   :sep:`|` :aspect:`Condition:` optional
+   :sep:`|` :aspect:`Type:` array
+   :sep:`|` :aspect:`Default:` '[]'
+   :sep:`|`
+
+   Headers to be sent with the response.
+
+$reasonPhrase
+   :sep:`|` :aspect:`Condition:` optional
+   :sep:`|` :aspect:`Type:` string
+   :sep:`|` :aspect:`Default:` ''
+   :sep:`|`
+
+   A reason for the given status code. If omitted, the default for the used status code will be used.
+
+.. code-block:: php
+
+   public function doSomethingAction(ServerRequestInterface $request): Response
+   {
+       // our previous computation
+
+       return new Response(json_encode(['result' => $result], 200, ['Content-Type' => 'application/json; charset=utf-8']))
```

Comment:
The class `JsonResponse` would fit better here.

Reply:
JsonResponse is internal API, tgat's why I didn't use it here.

Reply:
Mmh ok even though it's much cleaner.